tests/functional/test_locked_objects.py

```diff
@@ -9,190 +9,307 @@
-    return hostprovider.host_create(random_string())
-
-
-def _check_locked_object(obj: Union[Cluster, Service, Provider, Host]):
-    """
-    Assert that object state is 'locked' and action list is empty
-    """
-    obj.reread()
-    assert_state(obj=obj, state="locked")
-    assert (
-        obj.action_list() == []
-    ), f"{obj.__class__.__name__} action list not empty. {obj.__class__.__name__} not locked"
-
-
-def test_cluster_must_be_locked_when_action_running(prepared_cluster: Cluster):
-    with allure.step("Run action: lock-cluster for cluster"):
-        prepared_cluster.action(name="lock-cluster").run()
-    with allure.step("Check if cluster state is 'locked'"):
-        prepared_cluster.reread()
-        _check_locked_object(prepared_cluster)
-
-
-def test_run_new_action_on_locked_cluster_must_throws_exception(
-    prepared_cluster: Cluster,
-):
-    with allure.step("Run action: lock-cluster for cluster"):
-        prepared_cluster.action(name="lock-cluster").run()
-    with allure.step("Check that cluster state is 'locked'"):
-        _check_locked_object(prepared_cluster)
-
-
-def test_service_in_cluster_must_be_locked_when_cluster_action_running(
-    prepared_cluster: Cluster,
-):
-    with allure.step("Add service and run action: lock-cluster for cluster"):
-        added_service = prepared_cluster.service_add(name="bookkeeper")
-        prepared_cluster.action(name="lock-cluster").run()
-    with allure.step("Check if service state is 'locked'"):
-        _check_locked_object(added_service)
-
-
-def test_host_in_cluster_must_be_locked_when_cluster_action_running(
-    prepared_cluster: Cluster, host: Host
-):
-    with allure.step("Add host and run action: lock-cluster for cluster"):
-        prepared_cluster.host_add(host)
-        prepared_cluster.action(name="lock-cluster").run()
-    with allure.step("Check if host state is 'locked'"):
-        _check_locked_object(prepared_cluster)
-
+def host(host_provider: Provider) -> Host:
+    return host_provider.host_create(random_string())
 
-def test_host_must_be_locked_when_host_action_running(host):
-    with allure.step("Run host action: action-locker"):
-        host.action(name="action-locker").run()
-    with allure.step("Check if host state is 'locked'"):
-        _check_locked_object(host)
 
+@pytest.fixture()
+def complete_cluster(cluster: Cluster, host: Host):
+    cluster.host_add(host)
+    service = cluster.service_add(name="dummy")
+    cluster.hostcomponent_set((host, service.component(name="dummy")))
+    return cluster
+
+
+class TestClusterLock:
+    def test_lock_unlock(self, cluster, host):
+        """
+        Test that cluster locked when action running and unlocked when action ends
+        """
+        cluster.host_add(host)
+        assert_state(cluster, "created")
+        task = _lock_obj(cluster)
+        _check_that_object_is_locked(cluster)
+        task.wait()
+        assert_state(cluster, "created")
```

Comment:
Here and below in we assert state and actions list to ensure that object was locked.
But when we check obj unlock we assert only state.
Does it enough? Or should we add actions list assertion for unlock obj check too?

Reply:
I thought about it too

Reply:
Let's add it